src/player.cpp

```diff
@@ -6147,7 +6189,9 @@ void player::process_effects() {
 
             // Handle fatigue
             val = it.get_mod("FATIGUE", reduced);
-            if (val != 0) {
+            // Prevent ongoing fatigue effects while asleep.
+            // These are meant to change how fast you get tired, not how long you sleep.
+            if (val != 0 && !in_sleep_state()) {
```

Comment:
This also prevents extra fatigue from abusing meth.
Might be acceptable, but a solution that only affects long-term effects would be preferable.
Here's a relatively nice solution:
Since sleep is an effect, adding sleep as a `resist_effects` for cold and flu could allow treating sleep in a similar way flu med works here. Then `fatigue_min` would be `[1, 0]` to make it not drain fatigue when sleeping. It would also prevent the annoying cough wakeup that makes disease so much less fun. This would have a side effect of making flu med prevent fatigue drop, though.